MdeModulePkg/Library/DeviceStateLib/DeviceStateLib.c

```diff
@@ -13,30 +13,119 @@ MU_CHANGE: new file
 #include <Library/DeviceStateLib.h>
 #include <Library/DebugLib.h>
 
+CHAR8  *DEVICE_STATE_STRINGS[6] = {

```

Comment:
I suggest making `6` a macro defined value for clarity. For example, the earlier code defined `7` strings:

```c
#define DEVICE_STATE_SECUREBOOT_OFF_STRING             "SECURE_BOOT_OFF "
#define DEVICE_STATE_MANUFACTURING_MODE_STRING         "MANUFACTURING_MODE "
#define DEVICE_STATE_DEVELOPMENT_BUILD_ENABLED_STRING  "DEVELOPMENT_BUILD_ENABLED "
#define DEVICE_STATE_SOURCE_DEBUG_ENABLED_STRING       "SOURCE_DEBUG_ENABLED "
#define DEVICE_STATE_UNDEFINED_STRING                  "UNDEFINED "
#define DEVICE_STATE_UNIT_TEST_MODE_STRING             "UNIT_TEST_MODE "
#define DEVICE_STATE_PLATFORM_MODE_STRING              "OEM_DEFINED "
```

So, if the macro name indicated why these 6 were chosen and `DEVICE_STATE_PLATFORM_MODE_STRING` was excluded that would make it clear it was intentional and how to determine whether future strings should be added here or not.